backend/utils/_ipfs.js

```diff
@@ -186,20 +165,44 @@ async function getTextFn(gateway, hashAsBytes, timeoutMS) {
   return await response.text()
 }
 
-const getText = memoize(getTextFn, (...args) => args[1])
+const getTextCache = memoize(getTextFn, (...args) => args[1])
+
+/**
+ * Fetches content from IPFS and returns it as a string.
+ * Uses an in-memory cache.
+ *
+ * @param {string} gateway: URL of the IPFS gateway to use.
+ * @param {string} hashAsBytes: IPFS hash.
+ * @param {number} timeoutMS: Timeout in msec.
+ * @returns {Promise<string>}
+ * @throws
+ */
+async function getText(gateway, hashAsBytes, timeoutMS) {
+  try {
+    return getTextCache(gateway, hashAsBytes, timeoutMS)
```

Comment:
I vaguely recall dealing with this before in graphql(and [replacing memoize](https://github.com/OriginProtocol/origin/blob/master/packages/graphql/src/utils/memorize.js)).  Are you sure this will throw as you expect?

Reply:
Thank you for pointing this out. You were right, it did not throw an exception. Fixed and added unit test.